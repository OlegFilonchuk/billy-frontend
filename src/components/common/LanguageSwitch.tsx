import React, { FC } from 'react';

const LanguageSwitch: FC = () => {
  return <div>test</div>;
};

export default LanguageSwitch;
