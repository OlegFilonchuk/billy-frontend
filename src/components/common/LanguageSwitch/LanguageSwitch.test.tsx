import { fireEvent, RenderResult } from '@testing-library/react';
import React from 'react';
import renderWithI18n from '../../../utils/testUtills';
import LanguageSwitch from './LanguageSwitch';

describe('BackButton', () => {
  let ui: RenderResult;
  beforeEach(() => {
    ui = renderWithI18n(<LanguageSwitch />);
  });

  it('should render properly', () => {
    expect(ui.container).toMatchSnapshot();
  });

  it('should have english language button disabled', () => {
    const enButton = ui.getByText('en');
    expect(enButton).toBeDisabled();
  });

  it('should have russian language button enabled', () => {
    const ruButton = ui.getByText('ru');
    expect(ruButton).not.toBeDisabled();
  });

  it('should change buttons state after click', () => {
    const ruButton = ui.getByText('ru');
    fireEvent.click(ruButton);
    const enButton = ui.getByText('en');
    expect(ruButton).toBeDisabled();
    expect(enButton).not.toBeDisabled();
  });
});
