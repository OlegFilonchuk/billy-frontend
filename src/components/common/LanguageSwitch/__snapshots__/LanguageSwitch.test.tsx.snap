// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton should render properly 1`] = `
<div>
  <div
    class="flex"
  >
    <button
      class="transition duration-150 hover:shadow-lg focus:shadow-lg focus:ring ring-accent disabled:opacity-50 p-2 rounded inline-flex justify-center items-center    "
      type="button"
    >
      ru
    </button>
    <div
      class="w-px h-full bg-black"
    />
    <button
      class="transition duration-150 hover:shadow-lg focus:shadow-lg focus:ring ring-accent disabled:opacity-50 p-2 rounded inline-flex justify-center items-center    "
      disabled=""
      type="button"
    >
      en
    </button>
  </div>
</div>
`;
